File: mvpboilerplate/src/main/java/com/agit/lutluthfi/mvpboilerplate/base/IPlateBaseView.java
package com.agit.lutluthfi.mvpboilerplate.base;

import android.support.annotation.StringRes;
import android.view.View;

public interface IPlateBaseView {

    void showLoading();

    void hideLoading();

    boolean isLoading();

    boolean isNetworkConnected();

    void onError(String message);

    void onError(@StringRes int resId);

    void showMessage(String message);

    void showMessage(@StringRes int resId);

    void printLog(String tag, String message);

    void printLog(String tag, @StringRes int resId);

    void printLog(String tag, Throwable tr);

    void showSnackbar(String message, int duration, int textColor);

    void showSnackbar(@StringRes int resId, int duration, int textColor);

    void showSnackbarWithAction(String message, int duration, int textColor, String action, View.OnClickListener listener);

    void showSnackbarWithAction(@StringRes int resId, int duration, int textColor, String action, View.OnClickListener listener);

    void hideSnackbar();

    boolean isSnackbarShowing();
}

File: mvpboilerplate/src/main/java/com/agit/lutluthfi/mvpboilerplate/base/IPlateBaseView.kt
package com.agit.lutluthfi.mvpboilerplate.base

import android.support.annotation.StringRes
import android.view.View

interface IPlateBaseView {

    fun isNetworkConnected(): Boolean

    fun isSnackbarShowing(): Boolean

    fun isLoading(): Boolean

    fun showLoading()

    fun hideLoading()

    fun showMessage(message: String?)

    fun showMessage(@StringRes resId: Int)

    fun printLog(tag: String, message: String?)

    fun printLog(tag: String, @StringRes resId: Int)

    fun printLog(tag: String, tr: Throwable?)

    fun showSnackbar(message: String?, duration: Int, textColor: Int)

    fun showSnackbar(@StringRes resId: Int, duration: Int, textColor: Int)

    fun showSnackbarWithAction(message: String?,
                               duration: Int,
                               textColor: Int,
                               action: String,
                               listener: View.OnClickListener)

    fun showSnackbarWithAction(@StringRes resId: Int,
                               duration: Int,
                               textColor: Int,
                               action: String,
                               listener: View.OnClickListener)

    fun hideSnackbar()
}
